### ISSUE TITLE
Intel MKL FATAL ERROR: Cannot load libmkl_mc3.so or libmkl_def.so

### ISSUE THREAD

**Issue by jcohenadad** [neuropoly/spinalcordtoolbox#1597]
conversation with user (see email 2018-02-07):

Original email:
~~~
Avant de déposer une "issue" sur github, je me dis que tu as peut-être une réponse évidente à mon souci...
J'ai essayé d'installer SCT_v3.1.0 sur CentOS 7. L'installation se passe bien mais je n'arrive pas à tester toutes les fonctions :

sct_install : pas de problème sauf l'erreur suivante juste après l'installation de numpy :
WARNING conda.gateways.disk.permissions:make_writable(39): Error making path writable: /coconut/applis/bin/x86_64/sct_v3.1/python/bin/f2py.c~
OSError(5, 'Input/output error')
WARNING conda.gateways.disk.permissions:make_writable(39): Error making path writable: /coconut/applis/bin/x86_64/sct_v3.1/python/lib/python2.7/site-packages/numpy/__config__.py.c~
OSError(5, 'Input/output error')
WARNING conda.gateways.disk.permissions:make_writable(39): Error making path writable: /coconut/applis/bin/x86_64/sct_v3.1/python/lib/python2.7/site-packages/numpy/distutils/__config__.py.c~
OSError(5, 'Input/output error')
WARNING conda.gateways.disk.permissions:make_writable(39): Error making path writable: /coconut/applis/bin/x86_64/sct_v3.1/python/lib/python2.7/site-packages/numpy/distutils/site.cfg.c~
OSError(5, 'Input/output error')

sct_check_dependencies : aucune erreur

sct_testing : c'est là que ça se gâte :
--
Spinal Cord Toolbox (version: 3.1.0)
Running /coconut/applis/bin/x86_64/sct_v3.1/scripts/sct_testing.py

Path to testing data: /root/sct_testing_data/
Checking sct_analyze_lesion.........................[OK]
Checking sct_analyze_texture........................Intel MKL FATAL ERROR: Cannot load libmkl_mc3.so or libmkl_def.so.

Si je commente sct_analyze_texture dans sct_testing.py, pour tester les autres fonctions, toutes les autres fonctions passent le test *sauf*

Checking sct_process_segmentation...................Running /coconut/applis/bin/x86_64/sct_v3.1/scripts/sct_process_segmentation.py -a nurbs -r 1 -i /root/sct_testing_data/t2/t2_seg.nii.gz -no-angle 0 -v 1 -p centerline -use-image-coord 0 -vertfile ./label/template/PAM50_levels.nii.gz -ofolder /root/tmp.180205135209_936493/sct_process_segmentation_sct_testing_data_180205135639_571102/                        
Intel MKL FATAL ERROR: Cannot load libmkl_mc3.so or libmkl_def.so.                                                                       
Running /coconut/applis/bin/x86_64/sct_v3.1/scripts/sct_process_segmentation.py -a nurbs -r 1 -i /root/sct_testing_data/t2/t2_seg.nii.gz -no-angle 0 -v 1 -p length -use-image-coord 0 -vertfile ./label/template/PAM50_levels.nii.gz -ofolder /root/tmp.180205135209_936493/sct_process_segmentation_sct_testing_data_180205135639_571102/sct_process_segmentation_sct_testing_data_180205135641_291300/                  
Intel MKL FATAL ERROR: Cannot load libmkl_mc3.so or libmkl_def.so.                                                                       
[FAIL]

Checking sct_segment_graymatter.....................Running /coconut/applis/bin/x86_64/sct_v3.1/scripts/sct_segment_graymatter.py -ratio level -qc 0 -thr-sim 0.0005 -normalization 1 -i /root/sct_testing_data/t2s/t2s.nii.gz -r 1 -ref /root/sct_testing_data/t2s/t2s_gmseg_manual.nii.gz -w-levels 2.5 -ofolder ./ -denoising 1 -v 1 -res-type prob -p step=1,type=seg,algo=centermassrot,metric=MeanSquares,smooth=2,poly=0,iter=1:step=2,type=seg,algo=columnwise,metric=MeanSquares,smooth=1,iter=1 -s /root/sct_testing_data/t2s/t2s_seg.nii.gz -vertfile t2s/MNI-Poly-AMU_level_crop.nii.gz -w-coordi 0.0065
Intel MKL FATAL ERROR: Cannot load libmkl_mc3.so or libmkl_def.so.
[FAIL]

Tu aurais une idée d'où vient cette erreur sur libmkl, par hasard ? vous avez déjà rencontré ça ? j'espère que ce n'est pas lié aux mises à jour récentes des noyaux des systèmes CentOS, suite aux différentes failles de sécurité dont nous avons tous été arrosés... Pour info, mon noyau est
3.10.0-693.11.6.el7.x86_64
mais il y a le même pb sur le noyau le plus récent qui est 3.10.0-693.17.1.el7.x86_64.
~~~

Solution:
~~~

1) Can you run?

echo $LD_PRELOAD

2) If it empty, can you run?:

export LD_PRELOAD=/coconut/applis/bin/x86_64/sct_v3.1/python/lib/libmkl_core.so:/coconut/applis/bin/x86_64/sct_v3.1/python/lib/libmkl_sequential.so

followed by the sct_process_registration command that failed previously.


reference: https://software.intel.com/en-us/forums/intel-math-kernel-library/topic/300857#comment-1627042
~~~

**Comment by zougloub** [comment]
To be fixed by  #1703

**Comment by zougloub** [comment]
Fixed by #1703
